### ISSUE TITLE
NatNet not working anymore

### ISSUE THREAD

**Issue by dewagter** [paparazzi/paparazzi#3378]
![image](https://github.com/user-attachments/assets/02fa1d11-66af-404b-9e8d-b4b3a9d5f96c)
That's the output of the natnet2ivy.py script with the -v flag up. It is not able to properly identify the stream coming from motive and shuts down the interface. Perhaps the pose packets are different in the old SDK. Could we somehow make a flag (like the current "-o" flag) to allow backward compatibility? 
Probably we could also solve by purchasing a motive 3.x license but it's nice to have the capability of rolling back and use motive 2.x for any reason.

_Originally posted by @alessandro-mancinelli in https://github.com/paparazzi/paparazzi/issues/3249#issuecomment-2379062679_

**Comment by gautierhattenberger** [comment]
I agree that this is a problem. The natnet lib, especially in python, gives very little information about the possible issues and the differences between APIs. I don't think it even says that it should not be compatible.
One solution would be to analyze the errors in details and fix their code, if possible. Another option is just to have the two code side by side and select at startup the version you want. Since the callbacks prototypes and objects have changed, it still requires a little bit of work to make this not too ugly.

**Comment by fvantienen** [comment]
https://github.com/tudelft/paparazzi/commit/5b6730dece7755567a2ec8aeb1b5910eb065ea11#diff-dd2eab9474ada20e59ef6082f6fdf7449109f2e934baf22624910e75299eede7R93
This is not the proper fix, but makes it work with the old natnet. If we add something like this properly would be nice

**Comment by gautierhattenberger** [comment]
Here are the results:
- with `__nat_net_requested_version = [3,0,0,0]`, it is never working, it really needs to have `[0,0,0,0,]`. Even with that it is not starting every time, but this is a reported bug with this version, hopefully corrected with the "final" release of natnet coming soon.
- the `__server_version` works the same with `[3,0,0,0]` and `[0,0,0,0,]`
- we never have the `not natnet.connected()` case

**Comment by OpenUAS** [comment]
Tested https://github.com/tudelft/UnifiedMocapRouter and that works well (Ubuntu 20.04) no python dependence. Only it does depend on Boost lib, but that is installable via official Ubuntu repo. Proposal would be to include it as a Submodule.
It is well documented and also usable for e.g. PX4, ROS2 etc. Based on official libraries from Optitrack Motive manufacturer and works for Motive v1.x v2.x and v3.x

If you use the Master branch of Paparazzi or v7 and higher there is this new solution: https://github.com/tudelft/UnifiedMocapRouter

Follow the steps to build in the README.md or use this as quick n dirty way:

```
cd ~/paparazzi 
git submodule add https://github.com/tudelft/UnifiedMocapRouter.git sw/ext/unifiedmocaprouter
cd ~/paparazzi/sw/ext/unifiedmocaprouter/ && mkdir build && cd build
cmake -D'MOCAPS=optitrack' -D'AGENTS=console;ivy' .. && make
```
in your control panel file, usually add in the section of "programs" this line 

    <program name="UnifiedMoCap" command="sw/ext/unifiedmocaprouter/build/mocap-router"/>

Then this section will look something like:

```
  <section name="programs">
    <program name="Server" command="sw/ground_segment/tmtc/server"/>
    <program name="UnifiedMoCap" command="sw/ext/unifiedmocaprouter/build/mocap-router"/>

```

In Paparazzi Center open a Cyberzoo session, then click the +add program button and filter or look for "mocap", click it

The final line should then be modified to look similar like this:
```
/home/whomeveryouare/paparazzi/sw/ext/unifiedmocaprouter/build/mocap-router optitrack ivy --publish_frequency 10 --coordinate_system enu --coordinate_north far  --craft_noses right --streaming_ids 42 --ac_id 123
```

Do not forget to change
* whomeveryouare if it is not you username ;)
* the streaming_ids to the one you have on Motive 
* the ac_id to the one of your airframe.

**Comment by OpenUAS** [comment]
New Mocap router is added to repo. Note that this solution is using official libraries from Optitrack and only available for x86/amd64. for that a full open solution would be best.
